Code/Framework/AzCore/AzCore/Name/Name.cpp

```diff
@@ -224,11 +227,15 @@ namespace AZ
             return;
         }
 
-        if (s_staticNameListThread == AZStd::thread::id{})
+        const AZStd::thread::id thisThreadId = AZStd::this_thread::get_id();
+
+        if (s_staticNameListThread == InvalidThreadId)
         {
-            s_staticNameListThread = AZStd::this_thread::get_id();
+            s_staticNameListThread = thisThreadId;
         }
-        AZ_Assert(s_staticNameListThread == AZStd::this_thread::get_id(), "Attempted to construct a name literal on a different thread from the first initialized static name, this is unsafe");
+        AZ_Assert(s_staticNameListThread == thisThreadId,
+                  "Attempted to construct name literal '%.*s' on a different thread from the first initialized static name, this is unsafe",
+                  AZ_STRING_ARG(name));
```

Comment:
This assert should removed.
This makes it so that SetNameLiteral asserts if it is ever called on a thread that isn't the main one.
It is perfectly valid to link Names to a linked list on other threads

Reply:
ok, I'll remove it then.

Reply:
Actually, `s_staticNameListThread` was only used for this, so it can be removed altogether.